pandas/core/internals/blocks.py

```diff
@@ -1962,6 +1962,14 @@ class ObjectValuesExtensionBlock(ExtensionBlock):
     Series[T].values is an ndarray of objects.
     """
 
+    def diff(self, n: int, axis: int = 1) -> List["Block"]:
+        # Block.shape vs. Block.values.shape mismatch
```

Comment:
More technical debt from the 1D arrays inside 2D blocks :(

Reply:
Oh, and this is on ObjectValuesExtensionBlock, but is only useful for PeriodArray. IntervalArray is the only other array to use this, and doesn't implement `__sub__`. Is it worth creating a `PeriodBlock` just for this (IMO no)?

Reply:
Why is this only needed for ObjectValuesExtensionBlock, and not for ExtensionBlock?

Reply:
I suppose that in principle, we can hit this from ExtensionBlock.

We hit the problem when going from a NonConsolidatable block type (like period) to a consolidatable one (like object). In that case, the values passed to `make_block` will be 1d, but the block is expecting them to be 2d.

In practice, I think that for most EAs, `ExtensionArray.__sub__` will return another extension array. So while ExtensionBlock.diff isn't 100% correct for all EAs, I think trying to handle this case is not worth it. What do you think?

Reply:
> So while ExtensionBlock.diff isn't 100% correct for all EAs, I think trying to handle this case is not worth it

It certainly seems fine to ignore this corner case for now.